rest_auth/registration/views.py

```diff
@@ -91,31 +94,98 @@ def post(self, request, *args, **kwargs):
         return Response({'detail': _('ok')}, status=status.HTTP_200_OK)
 
 
-class SocialLoginView(LoginView):
-    """
-    class used for social authentications
-    example usage for facebook with access_token
-    -------------
-    from allauth.socialaccount.providers.facebook.views import FacebookOAuth2Adapter
+if 'allauth.socialaccount' in settings.INSTALLED_APPS:
+    from allauth.socialaccount import signals
+    from allauth.socialaccount.models import SocialAccount
+    from allauth.socialaccount.adapters import get_adapter
```

Comment:
I changed the `get_adapter` to import from the `socialaccount.adapters` path. I suspect this was the error here, but haven't tested E2E yet.